fe/fe-core/src/main/java/org/apache/doris/analysis/S3LoadStmt.java

```diff
@@ -0,0 +1,104 @@
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.analysis;
+
+import org.apache.doris.analysis.StorageBackend.StorageType;
+import org.apache.doris.common.AnalysisException;
+import org.apache.doris.common.DdlException;
+import org.apache.doris.datasource.property.constants.S3Properties;
+import org.apache.doris.datasource.property.constants.S3Properties.Env;
+import org.apache.doris.tablefunction.ExternalFileTableValuedFunction;
+import org.apache.doris.tablefunction.S3TableValuedFunction;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * For TVF solution validation.
+ */
+public class S3LoadStmt extends NativeInsertStmt {
+
+    public S3LoadStmt(LabelName label, List<DataDescription> dataDescList, BrokerDesc brokerDesc,
+            Map<String, String> properties, String comments) throws DdlException {
+        super(buildInsertTarget(dataDescList.get(0)),
+                label.getLabelName(), /*TODO(tsy): think about columns*/null,
+                buildInsertSource(dataDescList.get(0), brokerDesc), null);
+    }
+
+    private static InsertTarget buildInsertTarget(DataDescription dataDescription) {
+        final TableName tableName = new TableName(null, null, dataDescription.getTableName());
+        return new InsertTarget(tableName, dataDescription.getPartitionNames());
+    }
+
+    private static InsertSource buildInsertSource(DataDescription dataDescription, BrokerDesc brokerDesc)
+            throws DdlException {
+        final SelectList selectList = new SelectList();
+        // TODO(tsy): think about columns, now use `select *` way
+        final SelectListItem item = new SelectListItem(SelectListItem.createStarItem(null));
+        selectList.addItem(item);
+
+        final FromClause fromClause = new FromClause(
+                Collections.singletonList(buildTvfRef(dataDescription, brokerDesc))
+        );
+        final OrderByElement orderByElement = new OrderByElement(new StringLiteral(dataDescription.getSequenceCol()),
+                true, null);
+
+        final SelectStmt selectStmt = new SelectStmt(
+                selectList, fromClause, /*TODO(tsy): think about PRECEDING FILTER*/dataDescription.getWhereExpr(),
+                null, null,
+                Lists.newArrayList(orderByElement), LimitElement.NO_LIMIT
+        );
+        return new InsertSource(selectStmt);
+    }
+
+    private static TableRef buildTvfRef(DataDescription dataDescription, BrokerDesc brokerDesc) throws DdlException {
+        final Map<String, String> params = Maps.newHashMap();
+
+        final List<String> filePaths = dataDescription.getFilePaths();
+        Preconditions.checkState(filePaths.size() == 1, "there should be one and only file path");
```

Comment:
```suggestion
        Preconditions.checkState(filePaths.size() == 1, "there should be one and only file path");
```Preconditions.checkState(filePaths.size() == 1, "there should be only one file path");